Replace deprecated types in v0.4 of Julia, Uint->UInt, String->Abstra…

thomas-nilsson-irfu opened the pull request:
…ctString.

A minor disclaimer: I am not fully literate in Julia yet and therefor I might have made some mistake, but this should resolve some but not all the annoying warnings about deprecated types when using Matlab from the stable Julia v0.4+. (ie. As such no guarantees of functionality etc, are given by me or my employer).

timholy commented on src/engine.jl:
```diff
@@ -197,7 +197,7 @@ function make_getvar_statement(ex::Expr)
 end
 
-function _mget_multi(vs::Union(Symbol, Expr)...)
+function _mget_multi(vs::Union{Symbol, Expr}...)
```
You'll need to add `@compat` in front of this line if you want this package to still be usable for julia 0.3. Alternatively, we could bump the julia version in the REQUIRE file. Personally I think the latter is fine, if you think that's easier.

timholy commented:
Very nice!

If we just make sure we don't break 0.3, I'm fine with merging this. We don't have Travis turned on for this repository (because Travis doesn't have Matlab installed), but I ran the tests myself and they pass on 0.4 (as presumably @thomas-nilsson-irfu did, too).

thomas-nilsson-irfu commented:
Thank you, I was not aware of the @compat function.

I have just added it to all instances I changed that use the new Union{} instead of Union(), and I have run the Pkg.test("MATLAB") on Julia (but on v.0.4.1 only).

With best regards,
Thomas Nilsson

timholy commented:
Works on julia 0.3, too.

Thanks for the contribution!
